src/main/java/games/strategy/engine/lobby/server/userDB/DbUserController.java

```diff
@@ -0,0 +1,112 @@
+package games.strategy.engine.lobby.server.userDB;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import com.google.common.base.Preconditions;
+
+
+/**
+ * Interface for interacting with database user operations. For example, add user, delete user..
+ */
+/*
+ * TODO: datasource migration note, from derby (java DB) to MySQL
+ *  This class is currently set up to write and read from multiple data sources.
+ *  This is a migration tool, to go from one data source to another. When we switch over completely
+ *  we can simplify and write/read to the primary datasource directly.
+ */
+public class DbUserController implements UserDao {
+  private static final Logger logger = Logger.getLogger(DbUserController.class.getName());
+
+  private static final Collection<UserDaoPrimarySecondary> implementations = Arrays.asList(
+      new DerbyUserController(),
+      new SqlUserController()
+  );
+
+  /**
+   * Stores a convenience reference to the authoritative DAO implementation.
+   * We use this datasource for reading, we always write to it, any errors during read/write are considered
+   * critical (on the other hand secondary DAO read/write errors can be logged and ignored)
+   */
+  private static final UserDao primaryDao;
+
+
+  static {
+    primaryDao = determinePrimaryDataSource();
```

Comment:
Out of curiosity, is there a specific reason this has to be done in the class initializer and can't be inlined with the field initializer above?